src/Data/Collector.ts

```diff
@@ -270,6 +275,87 @@ const isReceiptRobust = async (
   return { success: true }
 }
 
+// Offline receipt verification
+const verifyReceiptOffline = async (
+  receipt: Receipt.ArchiverReceipt, 
+  executionGroupNodes: ConsensusNodeInfo[],
+  minConfirmations: number
+): Promise<{ success: boolean; newReceipt?: Receipt.ArchiverReceipt }> => {
+  const { signedReceipt, globalModification } = receipt
+
+  if (globalModification) {
+    const globalReceipt = signedReceipt as P2PTypes.GlobalAccountsTypes.GlobalTxReceipt
+    // Verify global modification receipt has enough valid signatures
+    const validSigners = new Set<string>()
+    
+    // Ensure signs array exists and is not empty
+    if (!globalReceipt.signs || !Array.isArray(globalReceipt.signs) || globalReceipt.signs.length === 0) {
+      return { success: false }
+    }
+
+    // Ensure tx exists and has required fields
+    if (!globalReceipt.tx || !globalReceipt.tx.address) {
+      return { success: false }
+    }
+
+    for (const sign of globalReceipt.signs) {
+      // Skip invalid signatures
+      if (!sign || !sign.owner) continue
+
+      const node = executionGroupNodes.find(n => n.publicKey === sign.owner)
```

Comment:
should this comparison be case insensitive?

Reply:
changed in updated code